### PR TITLE
Catching errors everywhere

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] Add exception handler to redux
- [x] Add exception handler to saga middleware
- [x] Add global error showing component to desktop / android / ios

### CODE REVIEW

File: desktop/package.json [hunk 1]
```diff
@@ -55,7 +55,7 @@
     "react-redux": "4.4.5",
     "react-tap-event-plugin": "1.0.0",
     "redux": "3.6.0",
-    "redux-saga": "0.11.1",
+    "redux-saga": "0.12.0",
```
Comment (chrisnojima) [inline desktop/package.json]: needed for onError in the create function

File: shared/global-errors/index.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+import React, {Component} from 'react'
+import {Box, Text, Icon, HOCTimers} from '../common-adapters'
+import {globalStyles, globalColors, globalMargins, transition} from '../styles'
+
+import type {Props} from './index'
+
+type Size = 'Closed' | 'Small' | 'Big'
+type State = {
+  size: Size,
+  cachedSummary: ?string,
```
Comment (chrisnojima) [inline shared/global-errors/index.js]: we keep the summary/details so when we close and wipe the values we don't see it disappear immediatley as its animating

File: shared/global-errors/index.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        </Box>
+        <Text type='BodySmall' style={detailStyle}>{details}</Text>
+      </Box>
+    )
+  }
+}
+
+const containerStyle = {
+  ...globalStyles.flexBoxColumn,
+  ...transition('max-heigh'),
```
Comment (chrisnojima) [inline shared/global-errors/index.js]: need to use max height so the animations can be animated, kinda annoying

File: shared/nav.desktop.js [hunk 4]
```diff
@@ -155,6 +156,7 @@ class Nav extends Component<void, Props, void> {
           <MetaNavigator
             tab={loginTab}
             rootComponent={Login} />
+          <GlobalError />
```
Comment (chrisnojima) [inline shared/nav.desktop.js]: talked to cecile and its correct for the error to go across the nav as its a global error and not a pane related thing

File: shared/store/configure-store.js [hunk 5]
```diff
@@ -40,6 +40,41 @@ for (const method in console) {
...
+      details = `${error.message}. ${error.stack}`
+    }
+  }
+
+  return {summary, details}
+}
+
+const crashHandler = (error) => {
+  if (theStore) {
+    // $FlowIssue
```
Comment (chrisnojima) [inline shared/store/configure-store.js]: if these comments aren't here i get a very strange flow error. maybe @MarcoPolo can help debug

Comment (MarcoPolo) [inline shared/store/configure-store.js]: `theStore` isn't typed?

Comment (chrisnojima) [inline shared/store/configure-store.js]: yup, weird. the errors were super obscure 🤕

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready to be looked at tnx

File: shared/global-errors/index.native.js [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+      cachedDetails: props.details,
+    }
+  }
+
+  componentWillMount () {
+    this._resetError(!!this.props.summary)
+  }
+
+  _onExpandClick = () => {
+    console.log('aaaaaa clicked')
```
Comment (MarcoPolo) [inline shared/global-errors/index.native.js]: ✂️

File: shared/global-errors/index.native.js [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+import type {Props} from './index'
+
+type Size = 'Closed' | 'Small' | 'Big'
+type State = {
+  size: Size,
+  cachedSummary: ?string,
+  cachedDetails: ?string,
+}
+
+class GlobalError extends Component<void, Props, State> {
```
Comment (MarcoPolo) [inline shared/global-errors/index.native.js]: This seems the same as the desktop, maybe make this a shared component and have platform specific stateless components for the render?

File: shared/nav.ios.js [hunk 8]
```diff
@@ -103,7 +104,7 @@ class Nav extends Component {
     )
   }
 
-  _renderContent (tab, module) {
+  _renderContent (tab, module, includeGlobalError) {
```
Comment (MarcoPolo) [inline shared/nav.ios.js]: this arg isn't used?

File: shared/global-errors/index.js (hunk 3 above)
Comment (MarcoPolo) [inline shared/global-errors/index.js]: that says `max-heigh` instead of `max-height`

File: shared/global-errors/index.native.js (hunk 7 above)
Comment (MarcoPolo) [inline shared/global-errors/index.native.js]: I'm talking more about the logic of hiding after a timeout and using state to keep the message, not the views. If you think that the mobile stuff will deviate from desktop thats fine too.

**Comment by MarcoPolo** [comment]
👍 % C

File: shared/nav.ios.js (hunk 8 above)
Comment (chrisnojima) [inline shared/nav.ios.js]: will remove. it used to be used but then i changed it

File: shared/global-errors/index.native.js (hunk 7 above)
Comment (chrisnojima) [inline shared/global-errors/index.native.js]: ahhh. hmmm. let's leave it for now

File: shared/global-errors/index.js (hunk 3 above)
Comment (chrisnojima) [inline shared/global-errors/index.js]: typo, i had this hardcoded to test it before using the transition class. thanks!

**Pull request merged**